### PR TITLE
Fix manual

### PR DESCRIPTION
**Pull request by pguenezan**
- update compiler version and args fetching from camisole
- add two documents

### CODE REVIEW

File: prologin/documents/templates/documents/index.html [hunk 1]
```diff
@@ -157,6 +157,19 @@ <h3>{% trans "Organization" %}</h3>
               </a>
             </td>
           </tr>
+          <tr>
+            <td>
+              <strong>{% trans "Launch" %}</strong>
```
Comment (seirl) [inline prologin/documents/templates/documents/index.html:162]: Why is this in "Organisation"?

File: prologin/documents/templates/documents/index.html [hunk 2]
```diff
@@ -157,6 +157,19 @@ <h3>{% trans "Organization" %}</h3>
...
+            </td>
+            <td>
+              <a href="{% url 'documents:organization:launch-mail-contestant' year %}" class="btn btn-default btn-xs">
+                <i class="fa fa-envelope"></i> {% trans "Mail to contestant" %}
```
Comment (seirl) [inline prologin/documents/templates/documents/index.html:166]: contestants

File: prologin/documents/templates/documents/index.html [hunk 3]
```diff
@@ -157,6 +157,19 @@ <h3>{% trans "Organization" %}</h3>
...
+              </a>
+              <a href="{% url 'documents:organization:launch-mail-school' year %}" class="btn btn-default btn-xs">
+                <i class="fa fa-envelope"></i> {% trans "Mail to school" %}
```
Comment (seirl) [inline prologin/documents/templates/documents/index.html:169]: schools

File: prologin/documents/urls.py [hunk 4]
```diff
@@ -40,6 +40,8 @@
 
 organization_patterns = [
     url(r'^enrollment-form$', documents.views.EnrollmentFormView.as_view(), name='enrollment-form'),
+    url(r'^launch-mail-contestant$', documents.views.LaunchMailToContestantView.as_view(), name='launch-mail-contestant'),
```
Comment (seirl) [inline prologin/documents/urls.py:43]: Why is this in organization?

File: prologin/problems/views.py [hunk 5]
```diff
@@ -454,27 +454,13 @@ def get_versions():
...
-            (Language.scheme, "gambit-c", "Gambit-C"),
-            (Language.vb, "mono-basic", "Mono Basic"),
-        ]
-        context['versions'] = [(lang, description, cached_versions.get(key))
-                               for lang, key, description in languages]
+        versions = []
+        for language in cached_versions.get('languages').values():
+            name = language.get('name')
+            program = list(language.get('programs').keys())[0]
+            version = language.get('programs').get(program).get('version')
```
Comment (seirl) [inline prologin/problems/views.py:461]: Why do you use get() if it just crashes when get() returns None?

File: prologin/problems/views.py [hunk 6]
```diff
@@ -454,27 +454,13 @@ def get_versions():
...
-            (Language.python, "python", "CPython"),
-            (Language.scheme, "gambit-c", "Gambit-C"),
-            (Language.vb, "mono-basic", "Mono Basic"),
-        ]
-        context['versions'] = [(lang, description, cached_versions.get(key))
-                               for lang, key, description in languages]
+        versions = []
+        for language in cached_versions.get('languages').values():
+            name = language.get('name')
+            program = list(language.get('programs').keys())[0]
```
Comment (seirl) [inline prologin/problems/views.py:460]: Why do you only take the first compiler? All of them are important.

**Review by seirl — CHANGES_REQUESTED** [review]

File: prologin/problems/templates/problems/manual.html [hunk 7]
```diff
@@ -110,22 +110,24 @@ <h3>{% trans "Hardware architecture" %}</h3>
...
-{# TODO: retrieve from VM when it's available #}
-<dt>C</dt><dd><code>-Wextra -Wall -O2 -std=c11 -lm</code></dd>
-<dt>C++</dt><dd><code>-Wextra -Wall -O2 -std=c++14 -lm</code></dd>
-<dt>C#</dt><dd><code>-optimize+</code></dd>
-<dt>F#</dt><dd><code>-O</code></dd>
-<dt>Haskell</dt><dd><code>-dynamic -O2</code></dd>
-<dt>OCaml</dt><dd><code>-w A</code></dd>
-<dt>Pascal</dt><dd><code>-XD -Fainitc</code></dd>
+{% for lang, description, version, args in versions %}
+  {% if args|length != 0 %}
```
Comment (zopieux) [inline prologin/problems/templates/problems/manual.html:122]: ``if args``?

File: prologin/problems/templates/problems/manual.html [hunk 8]
```diff
@@ -110,22 +110,24 @@ <h3>{% trans "Hardware architecture" %}</h3>
...
+    <dt>{{ lang }}</dt><dd>
+    <code>
+    {% for arg in args %}
+      {{ arg }}
+    {% endfor %}
+    </code/></dd>
+  {% endif %}
+{% endfor %}
```
Comment (zopieux) [inline prologin/problems/templates/problems/manual.html:130]: Okay so you get this from `/languages`:
```json
 "brainfuck": {
            "name": "Brainfuck",
            "programs": {
                "esotope-bfc": {
                    "opts": [],
                    "version": null
                },
                "gcc": {
                    "opts": [
                        "-std=c11",
                        "-Wall",
                        "-Wextra",
                        "-O2",
                        "-lm"
                    ],
                    "version": "7.2.0"
                },
                "python2": {
                    "opts": [
                        "-S",
                        "/usr/sbin/esotope-bfc",
                        "-v",
                        "-fc"
                    ],
                    "version": "2.7.14"
                }
            }
        },
```
so eg. for Brainfuck we want this:

>Brainfuck:
>  * esotope-bfc
>  * gcc 7.2.0 with `-std=c11 -Wall -Wextra -O2 -lm`
>  * python2 2.7.14 with `-S /usr/bin/esotope-bfc -v -fc`

**Review by zopieux — CHANGES_REQUESTED** [review]

**Comment by zopieux** [comment]
Please split the mailing stuff in a new PR, as the training manual stuff was superseded by 324bcac.

**Pull request closed without merging**